src/xma/include/lib/xma_session.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XMA_SESSION_H_
+#define _XMA_SESSION_H_
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include "app/xmahw.h"
+#include "plg/xmasess.h"
+
+namespace xma_core {
+namespace plg {
+
+/**
+ * class session : abstraction for xma session
+ */
+class session
+{
+protected:
```

Comment:
I am wary about protected.  But occasionally it is needed and make sense, but I don't have the context here.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rnr-protected-data

Reply:
Thought about it. Use of private and getter/setter is better. Done